### PR TITLE
feat: fliter-by

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/common/Search/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    ml: 1,
+    flex: 1,
+  },
+  divider: { height: 20, m: 0.5 },
+}
+
+// Component
+const Search: FC<SearchProps> = (props) => {
+  const { facetName, searchTerm, onSearch, childInputRef, showClearButton = false } = props
+  const placeholder = `Search ${facetName}`
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/common/Search/Search.tsx]: yes, created prop

File: components/common/Search/Search.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  const { t } = useTranslation('common')
+  const searchIconAriaLabel = t('search-icon')
+  const searchInputAriaLabel = t('search-input')
+  const clearSearchAriaLabel = t('clear-search')
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    onSearch(event.target.value)
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/common/Search/Search.tsx]: SearchBar is a stateless component, hence currently searchTerm is handled by a parent, but if needed we can convert it into a stateful component. Let me know your thought.

[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/common/Search/Search.tsx]: SearchBar is a stateless component, hence currently searchTerm is handled by a parent, but if needed we can convert it into a stateful component. Let me know your thought.

File: components/common/Search/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  },
+  inputBase: {
+    ml: 1,
+    flex: 1,
+  },
+  divider: { height: 20, m: 0.5 },
+}
+
+// Component
+const Search: FC<SearchProps> = (props) => {
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/common/Search/Search.tsx]: done

[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/common/Search/Search.tsx]: done

File: components/filter-by/Facet/Facet.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+// Component
+const Facet: FC<FacetProps> = (props) => {
+  const { label, values = [] } = props
+
+  const { t } = useTranslation('common')
+  const viewMore = t('view-more')
+  const viewLess = t('view-less')
+
+  const childInputRef = useRef<HTMLInputElement | undefined>(null)
+  const defaultNoOfItemsToShow = 6
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/filter-by/Facet/Facet.tsx]: done

File: components/filter-by/Facet/Facet.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+import { Accordion, AccordionDetails, AccordionSummary, Button, Typography } from '@mui/material'
+import { grey } from '@mui/material/colors'
+import { Theme } from '@mui/material/styles'
+import { Box } from '@mui/system'
+import { useTranslation } from 'next-i18next'
+
+import Search from '../../common/Search/Search'
+import FacetItemList from '../FacetItemList/FacetItemList'
+
+import { Facet as FacetType, FacetValue, Maybe } from '@/lib/gql/types'
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/filter-by/Facet/Facet.tsx]: done

File: components/filter-by/Facet/Facet.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/filter-by/Facet/Facet.tsx]: done

File: components/filter-by/FacetItem/FacetItem.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  // URL params will be used by PLP page
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const isChecked = event.target.checked
+    let filters = router?.query?.filters || ''
+    if (typeof filters !== 'string') return
+
+    filters = isChecked ? filters + `${filterValue},` : filters.replace(`${filterValue},`, '')
+    router?.push({
+      pathname: router.pathname,
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/filter-by/FacetItem/FacetItem.tsx]: created hook and used it

[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/filter-by/FacetItem/FacetItem.tsx]: created hook and used it

File: components/filter-by/Facet/Facet.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/filter-by/Facet/Facet.tsx]: done

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]
looks good!

File: components/common/QuantitySelector/QuantitySelector.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import React, { FC } from 'react'
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/common/QuantitySelector/QuantitySelector.tsx:1]: resolved

File: components/filter-by/FacetItem/FacetItem.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    const action = isChecked ? 'add' : 'remove'
+
+    updateRoute(filterValue, action)
+    setChecked(isChecked)
+  }
+
+  // To mark FacetItem(checkbox) checked on Page reolad/refresh
+  useEffect(() => {
+    const filters = router?.query?.filters || ''
+    if (filters.includes(filterValue)) setChecked(true)
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/filter-by/FacetItem/FacetItem.tsx:48]: resolved
const [checked, setChecked] = useState<boolean>(isApplied)

[the remark this answers is not in the document]
Comment (kibo-kevinwatts) [reply components/filter-by/FacetItem/FacetItem.tsx:48]: do we still need useEffect check if pulling from "isApplied" for initial state?

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**